rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -186,10 +186,8 @@ def build_form_payload(parameters, example)
         # Rather that serializing with the appropriate encoding (e.g. multipart/form-data),
         # Rails test infrastructure allows us to send the values directly as a hash
         # PROS: simple to implement, CONS: serialization/deserialization is bypassed in test
-        tuples = parameters
-          .select { |p| p[:in] == :formData }
-          .map { |p| [p[:name], example.send(p[:name])] }
-        Hash[tuples]
+        form_data_param = parameters.select { |p| p[:in] == :formData }.first
```

Comment:
That `.first` worries me. Doesn't this just work with the 1st form parameter? 
```
{ name: 'f1', in: :formData, type: :string },
{ name: 'f2', in: :formData, type: :string }
```

Wouldn't `f2` be excluded?

The test seemed to test for 2 parameters in FormData, but now it only tests one. Could you return the test to have 2 form parameters to make sure it works with more that 1 form parameter?

Reply:
@sergioisidoro Thanks for your feedback! Yes, that's exactly how it would work. I have explained the reasoning in the "solution" paragraph. `body` parameters [work the same way](https://github.com/rswag/rswag/blob/master/rswag-specs/lib/rswag/specs/request_factory.rb#L235).
Another solution would be to make [swagger_formatter](https://github.com/rswag/rswag/blob/master/rswag-specs/lib/rswag/specs/swagger_formatter.rb#L58) accept multiple parameters because now it accepts only the first one and logic in these places diverge.

I agree this solution isn't great and is more of a quick fix. Currently, to work around this issue we define the first formData param just for `SwaggerFormatter` and then redefine the same schema in separate params to make `RequestFactory` work.

Reply:
Ok, let me take a bit longer to look into this, and see if I can figure out the historical background behind the body para. My guess is that it is also no supposed to be like that 🤔

Reply:
I don't think the current implementation is actually working for people. #348 is a long thread of people resigned to either having the integration test work, or their generated documentation being correct. I was about to submit pretty much this same PR but happy to see it's already been done.

Reply:
I have come up against this issue recently. And I am very keen to solve it, happy to work on it if we get more ideas on the best solution.